src/tiramisu_function.cpp

```diff
@@ -116,6 +116,112 @@ isl_union_map *tiramisu::function::compute_dep_graph() {
     return result;
 }
 
+const std::map<std::string, tiramisu::buffer *> tiramisu::function::get_mapping() const
+{
+  return this->mapping;
+
+}
+
+void  tiramisu::function::add_mapping(std::pair<std::string,tiramisu::buffer *> p)
+{
+  this->mapping.insert(p);
+
+}
+
+	
+const int  &function::Automatic_communication(tiramisu::computation* c1,tiramisu::computation* c2 ) const
+{
+    std::map<std::string, tiramisu::buffer*> buff = this->get_buffers();
+    std::map<std::string, tiramisu::buffer*> mp = this->mapping;
+    std::map<std::string, tiramisu::buffer*>::iterator it ;
+    std::string name, cpt_name;
+    int i = 1;
+    char tag;  
+    tiramisu::computation* first_cpt = c1;
+    tiramisu::computation* last_cpt = c2;
+    for (it = buff.begin(); it != buff.end(); ++it)
```

Comment:
I can't understand this loop, can you please add some documentation so that I can follow it?

Reply:
buff is the list of the buffers, i'm looping over this list. 
for each buffer i verify if it is not temporary, because i need only input and output buffers.
I have to cases tag = c which means that  i ll put it in constant memory 
                          tag = g which means that i ll put it in global memory

if tag = c 
   i ll verify if this buffer exists in the mapping mp (a map of correspondance cpu/gpu beffer)
   i'll only have to check if it's an input because i'll not have to copy back a constant 
   i'll verify if the user didn't copy manually by checking auto-trans which is modified in the function -- tag_gpu_constant () ;
then i'll create the memcpy computation if it s not done.

if tag = g which is the default case 
  i verify if it's an input and if it is the mapping ... the same processes only here i'll have two cases the input and the output

Reply:
Sorry, I don't mean you should explain to me. You should add this explanation in the code. But it should be a brief description.

Reply:
yeah i will add it